backend/inspirehep/editor/authorlist_utils.py

```diff
@@ -0,0 +1,345 @@
+import re
+
+from inspire_schemas.api import LiteratureBuilder
+from refextract.documents.pdf import replace_undesirable_characters
+
+re_emptyline = re.compile(r"\n\s*\n", re.UNICODE)
+re_hyphens = re.compile(
+    r"(\\255|\u02D7|\u0335|\u0336|\u2212|\u2013|\u002D|\uFE63|\uFF0D)", re.UNICODE
+)
+re_multiple_space = re.compile(r"\s{2,}", re.UNICODE)
+re_potential_key = re.compile(r"^(?:\d|[^\w.'-])+$", re.UNICODE)
+re_trailing_nonword = re.compile(r"((?:\d|[^\w,.'-])+ )", re.UNICODE)
+re_symbols = re.compile(r"[^\w ]", re.UNICODE)
+
+
+def split_id(word):
+    """
+    Separate potential aff-ids .
+    E.g.: '*12%$' -> ['*', '12' '%', '$']
+    """
+
+    aff_ids = []
+
+    symbols = re_symbols.findall(word)
+    if symbols:
+        aff_ids += symbols
+        for rest in re_symbols.split(word):
+            if rest:
+                aff_ids.append(rest)
+    else:
+        aff_ids.append(word)
+    return aff_ids
+
+
+def cleanup_affiliations_text(text):
+    text_with_replaced_chars = (
+        text.replace(",", " , ").replace("\n", " , ").replace(" and ", " , ") + " "
+    )
+    text_with_trailed_non_word = re_trailing_nonword.sub(
+        r" \1", text_with_replaced_chars
+    )
+    return re_multiple_space.sub(" ", text_with_trailed_non_word)
+
+
+def add_author_affiliation(affiliations, affiliation_key, author_affiliations):
+    if affiliation_key not in list(affiliations):
+        return
+    author_affiliations.append(affiliations[affiliation_key])
+    return affiliation_key
+
+
+def check_affiliation_key_type(aff_keys, warnings):
+    if not aff_keys:
+        warnings.append("Found no affiliations (empty line needed)")
+        return ""
+    if aff_keys[0].isalpha():
+        return "alpha"
+    if aff_keys[0].isdigit():
+        return "digit"
+    warnings.append("CAUTION! Using symbols (# and stuff) as aff-IDs.")
+    return "symbol"
+
+
+def create_author_fullname(author_names, warnings):
+    fullname = " ".join(author_names)
+    if len(author_names) == 1:
+        warnings.append("Author without firstname: %s" % fullname)
+    return fullname
```

Comment:
IMHO it should be cleaner to not use warnings from outside, but create here a separate warnings instance and return it together with fullname. When you can `extend` outside warnings list.